x-pack/plugins/alerting/server/task_runner/create_execution_handler.ts

```diff
@@ -71,6 +73,25 @@ export function createExecutionHandler<
     alertExecutionStore,
     alertId,
   }: ExecutionHandlerOptions<ActionGroupIds | RecoveryActionGroupId>) => {
+    const hasReachedTheExecutableActionsLimit = (): boolean =>
```

Comment:
Would it make sense to pull these out into a utility function so we can unit test them?

Reply:
Done.
I converted AlertsExecutionStore into a class and moved all those functions into it. Added unit tests as well :)